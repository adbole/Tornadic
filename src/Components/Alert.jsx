import PropTypes from 'prop-types'
import { Widget } from "./BaseComponents";

const Alert = (props) => (
    <Widget className={"alert " + props.type}>
        <h2>{props.name}</h2>
        <p>{props.message}</p>
    </Widget>
)
Alert.propTypes = {
    type: PropTypes.oneOf(['warning', 'watch', 'advisory', 'special']).isRequired,
    name: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
    moreExist: PropTypes.bool
}

export default Alert;