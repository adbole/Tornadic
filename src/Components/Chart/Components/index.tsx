export { default as Area } from "./Area"
export { default as Axes } from "./Axes"
export { default as Bar } from "./Bar"
export { default as Line } from "./Line"
export { default as NowReference } from "./NowReference"
export { default as Tooltip, useTooltip } from "./Tooltip"