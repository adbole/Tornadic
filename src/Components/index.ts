export { default as Alert } from "./Alert";
export { default as Daily } from "./Daily";
export { default as HazardLevel } from "./HazardLevel";
export { default as Hourly } from "./Hourly";
export * from "./Input";
export { default as MessageScreen } from "./MessageScreen";
export { default as Chart } from "./Modals/Chart";
export { default as Modal } from "./Modals/Modal";
export * from "./Modals/Modal";
export { default as Now } from "./Now";
export { default as Pressure } from "./Pressure";
export { default as Radar } from "./Radar";
export { default as SimpleInfoWidget } from "./SimpleInfoWidget";
export { default as SkeletonWidget } from "./SkeletonWidget";
export { default as SunTime } from "./SunTime";
export { default as Widget } from "./Widget";
export { default as Wind } from "./Wind";
