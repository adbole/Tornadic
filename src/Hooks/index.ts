export { default as useAnimation } from "./useAnimation";
export { default as useBooleanState } from "./useBooleanState";
export { default as useNullableState } from "./useNullableState";