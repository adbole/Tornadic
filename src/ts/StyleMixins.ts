import { css } from "@emotion/react";


export const mediaQueries = {
    large: "@media screen and (min-width: 1400px)",
    mediumMin: "@media screen and (min-width: 1100px)",
    mediumMax: "@media screen and (max-width: 1100px)",
    smallMin: "@media screen and (min-width: 800px)",
    smallMax: "@media screen and (max-width: 800px)",
} as const;

export const vars = {
    background: "var(--background)",
    backgroundLayer: "var(--background-layer)",
    borderRadius: "var(--border-radius)",
    inputBorderRadius: "var(--input-border-radius)",

    svgSize: "var(--svg-size)",

    primary: "var(--primary)",

    zLayer1: "var(--z-layer-1)",
    zLayer2: "var(--z-layer-2)",
} as const;

export const varNames= {
    background: "--background" as any,
    backgroundLayer: "--background-layer" as any,
    borderRadius: "--border-radius" as any,
    inputBorderRadius: "--input-border-radius" as any,

    svgSize: "--svg-size" as any,

    primary: "--primary" as any,

    zLayer1: "--z-layer-1" as any,
    zLayer2: "--z-layer-2" as any,
} as const;

export const alertColors = {
    warning: {
        background: "#C31700",
        foreground: "white",
    },
    watch: {
        background: "#FFF501",
        foreground: "black",
    },
    advisory: {
        background: "#FF9431",
        foreground: "black",
    },
    statement: {
        background: "#8F00FF",
        foreground: "white",
    },
    none: {
        background: "gray",
        foreground: "black",
    },
};


export const darkBackBlur = css({
    backgroundColor: "rgba(0, 0, 0, 0.75)",
    backdropFilter: "blur(10px)",
    color: "white",
});

export const centerFlex = css({
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
});

export const interactable = css({
    transition: "filter 0.3s ease",
    filter: "brightness(100%)",

    "&:hover, &:focus": { cursor: "pointer" },

    "&:active, &:disabled": { filter: "brightness(70%)" },
});

