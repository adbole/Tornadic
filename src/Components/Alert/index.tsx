import React from "react";
import booleanPointInPolygon from "@turf/boolean-point-in-polygon";

import { useBooleanState, usePermission } from "Hooks";

import { useWeather } from "Contexts/WeatherContext";

import AlertModal from "Components/Modals/Alert";

import type NWSAlert from "ts/NWSAlert";

import AlertWidget, { AlertInformation, ExcessAlerts } from "./style";


const getPriorityAlert = (alerts: NWSAlert[]) =>
    alerts.reduce(
        (highest, next) => (next.priority < highest.priority ? next : highest),
        alerts[0]
    );

function Alert({ noNotify = false }) {
    const { alerts: unfilteredAlerts, point } = useWeather();
    const notiPermission = usePermission("notifications")

    const [modalOpen, showModal, hideModal] = useBooleanState(false);

    const alerts = unfilteredAlerts.filter(alert => {
        //Point and NWSAlert include the necessary GeoJSON properties for this to work so conversion to any is okay
        const inBounds = alert.hasCoords() && booleanPointInPolygon(point as any, alert as any)
            
        return alert.get("affectedZones").includes(point.properties.forecastZone) || inBounds
    });

    const previousAlertTime = React.useRef(new Date(alerts[0]?.get("sent") ?? 0));

    React.useEffect(() => {
        if (noNotify || alerts.length === 0 || notiPermission !== "granted") return;

        const newAlerts = alerts.filter((alert) => new Date(alert.get("sent")) > previousAlertTime.current);

        if (newAlerts.length > 0) {
            const alert = getPriorityAlert(newAlerts);

            const additionalAlerts = newAlerts.length > 1 ? `\n+${newAlerts.length - 1} more alert(s)` : "";

            new Notification(`${alert.get("event")}`, {
                body: `Issued: ${alert.get("sent")}\nExpires: ${alert.get("expires") ?? alert.get("ends")}${additionalAlerts}`,
            });

            previousAlertTime.current = new Date(newAlerts[0].get("sent"));
        }

    //This effect should only run if alerts changes, noNotify and notiPermission are excluded as changing them could 
    //cause unnecessary notifications to be sent annoying the user.
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [alerts]);

    if (!alerts.length) return null;

    const alert = getPriorityAlert(alerts);

    return (
        <>
            <AlertWidget
                type={alert.getAlertCSS()}
                isTemplate
                size="widget-wide"
                onClick={showModal}
            >
                <AlertInformation>
                    <h1>{alert.get("event")}</h1>
                    <p>
                        <em>Issued:</em> {alert.get("sent")}
                    </p>
                    <p>
                        <em>Until:</em> {alert.get("ends") ?? alert.get("expires")}
                    </p>
                </AlertInformation>

                {alerts.length > 1 && (
                    <ExcessAlerts>+{alerts.length - 1} more alert(s)</ExcessAlerts>
                )}
            </AlertWidget>
            <AlertModal alerts={alerts} isOpen={modalOpen} onClose={hideModal} />
        </>
    );
}

export default Alert;
