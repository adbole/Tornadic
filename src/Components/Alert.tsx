import React from 'react';
import { AlertType, WeatherData } from '../ts/WeatherData';
import { NWSAlert, useWeather } from './Contexes/WeatherContext';
import { Modal, useModal } from './Contexes/ModalContext';
import { Widget, WidgetSize } from './SimpleComponents';

/**
 * Takes an ISO8601 string and converts it to a nice readable string in the format weekday, month, day, hour:minute AM/PM timezone
 * @param iso The ISO8601 string to convert
 * @returns The now nice readable string obtained from conversion
 */
function GetTimeString(iso: string) {
    return new Date(iso).toLocaleTimeString("en-us", {weekday:"short", month:"short", day:"numeric", hour12:true, hour:"numeric", minute:"numeric", timeZoneName:"short"});
}

function ToAlertCSS(alert: AlertType) {
    switch(alert) {
        case AlertType.WARNING:
            return "alert-warning";
        case AlertType.WATCH:
            return "alert-watch";
        case AlertType.ADVISORY:
            return "alert-advisory";
        default:
            return "alert-none";
    }
}

const AlertDisplay = ({alertData, className, onClick} : {alertData: NWSAlert, className: string, onClick: React.MouseEventHandler<HTMLDivElement>}) => (
    <Widget size={WidgetSize.WIDE} id="alert" className={className} onClick={onClick}>
        <h2>{alertData.properties.event}</h2>
        <p>{alertData.properties.event} until {GetTimeString(alertData.properties.ends ?? alertData.properties.expires)}</p>
    </Widget>
);

const Alert = () => {
    const alertData = useWeather().alerts;
    const modals = useModal();

    //If no alerts are active then don't display this component.
    if(!alertData.length) return <></>;

    //Determine which alert should be shown. Warnings get highest priority with the lastest being first. 
    let alertToShow = alertData[0];
    let highestAlertType = WeatherData.GetAlertType(alertData[0]);
    let newestAge = new Date(alertData[0].properties.sent);

    for(let i = 1; i < alertData.length; ++i) {
        const ithType = WeatherData.GetAlertType(alertData[i]);
        const sentDate = new Date(alertData[i].properties.sent);

        //Replace the alertToShow if its severity is lower
        if(highestAlertType < ithType) {
            alertToShow = alertData[i];
            highestAlertType = ithType;
            newestAge = sentDate;
        }
        //When two alerts are the same severity, we want the youngest one
        else if(highestAlertType === ithType && sentDate > newestAge) {
            alertToShow = alertData[i];
            highestAlertType = ithType;
            newestAge = sentDate;
        }
    }

    const onClickHandler = () => modals.showModal(alertData.length > 1 ? <AlertSelectionModal alertData={alertData}/> : <AlertModal alert={alertData[0]}/>);

    return <AlertDisplay alertData={alertToShow} className={ToAlertCSS(highestAlertType)} onClick={onClickHandler}/>;
};

export default Alert;

const AlertSelectionModal = ({alertData}: {alertData: NWSAlert[]}) => {
    const modals = useModal();

    //Memoize the component to prevent unnecessary operations
    const memoizedComponent = React.useMemo(() => (
        <Modal modalTitle={alertData.length + " Weather Alerts"} modalTitleClass={"alert-none"} id="alert-list-modal">
            {
                alertData.map((alert, key) => {
                    //Inclues override to onClose to reopen the alert selection modal
                    const onClickHandler = () => modals.showModal(<AlertModal alert={alert} onClose={() => modals.showModal(memoizedComponent)}/>);

                    return <AlertDisplay key={key} alertData={alert} className={ToAlertCSS(WeatherData.GetAlertType(alert))} onClick={onClickHandler}/>;
                })
            }
        </Modal>
    ), [alertData]);
    

    return memoizedComponent;
};

/**
 * Display the given NWSAlert as a modal to get more in depth information from.
 */
export const AlertModal = (props: {alert: NWSAlert} & React.DialogHTMLAttributes<HTMLDialogElement>) => {
    const {alert, ...excess} = props;

    const alertData = alert.properties;
    const alertType = ToAlertCSS(WeatherData.GetAlertType(alert));
    
    return (
        <Modal modalTitle={alertData.event} modalTitleClass={alertType} id="alert-modal" {...excess}>
            <p><em>Issuing Office:</em> {alertData.senderName}</p>
            <p><em>Issued:</em> {GetTimeString(alertData.sent)}</p>
            <p><em>Effective:</em> {GetTimeString(alertData.effective)}</p>
            <p><em>Ends:</em> {GetTimeString(alertData.ends ?? alertData.expires)}</p>
            <hr/>
            {
                alertData.instruction && 
                <>
                    <h3>Instructions</h3>
                    <p>{alertData.instruction}</p>
                    <hr/>
                </>
            }
            {
                alertData.description.split("*").map((string, index) => {
                    if(!string.length) return <React.Fragment key={index}/>;
        
                    return <p key={index}>* {string}</p>;
                })    
            }
            <hr/>
            <h3>Affected Areas</h3>
            <p>{alertData.areaDesc}</p>
        </Modal>
    );
};