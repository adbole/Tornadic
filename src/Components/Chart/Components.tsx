import { AreaChart, BarChart, LineChart, TooltipProps } from "recharts";
import { CategoricalChartProps } from "recharts/types/chart/generateCategoricalChart";
import { NameType } from "recharts/types/component/DefaultTooltipContent";

import { UV_MAX_VALUES } from "ts/Constants";

import { ChartViews, DataPoint } from ".";

function getUnit(data: DataPoint, defaultUnit: React.ReactNode) {
    const { property, primaryKey } = data;

    if(property === ChartViews.UV_Index) {
        if (primaryKey <= UV_MAX_VALUES.LOW) return " Low"; 
        else if (primaryKey <= UV_MAX_VALUES.MODERATE) return " Moderate";
        else if (primaryKey <= UV_MAX_VALUES.HIGH) return " High";
        else if (primaryKey <= UV_MAX_VALUES.VERY_HIGH) return " Very High";
        else return " Extreme"; 
    }

    return defaultUnit;
}
  
//Customized tooltip to display primary and secondary data with labels if applicable
export const CustomTooltip = ({ active, payload }: TooltipProps<number, NameType>) => {
    if(!active || !payload || !payload.length) return null;

    const fixDecimal = (property: ChartViews, value: number): string | number => 
        (property === ChartViews.Precipitation || property === ChartViews.Pressure) ? value.toFixed(2) : value.toFixed(0);

    const data: DataPoint = payload[0].payload;
    let unit = getUnit(data, payload[0].unit);
    const label = (data.property === ChartViews.Temperature && "Feels: ") || (data.property === ChartViews.Windspeed && "Gust: ");


    return (
        <div className="chart-tooltip">
            <h1>{fixDecimal(data.property, data.primaryKey)}{unit}</h1>
            {data.secondaryKey && <p>{label}{fixDecimal(data.property, data.secondaryKey)}{unit}</p>}
        </div>
    );
};

type ChartDisplayProps = {
    children: React.ReactNode,
    property: ChartViews
} & CategoricalChartProps

//Provides the proper chart to display based on the HourlyProperty passed
export const ChartDisplay = (props: ChartDisplayProps) => {
    const { children, property, ...excess } = props;

    let ChosenChart = BarChart;

    switch(property) {
        case ChartViews.Precipitation:
            ChosenChart = BarChart;
            break;
        case ChartViews.Temperature:
        case ChartViews.UV_Index:
            ChosenChart = AreaChart;
            break;
        default:
            ChosenChart = LineChart;
    }

    return (
        <ChosenChart {...excess}>
            {children}
        </ChosenChart>
    );
};