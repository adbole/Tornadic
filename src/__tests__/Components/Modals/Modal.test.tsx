import { act, fireEvent, render, screen } from "@testing-library/react";

import { useBooleanState } from "Hooks";

import { Modal, ModalContent, ModalTitle } from "Components";
import Dialog from "Components/Modals/Modal/style";

import { mediaQueries, vars } from "ts/StyleMixins";


const onClose = vi.fn();

function TestComponent({ defaultOpen = false }: { defaultOpen?: boolean }) {
    const [isOpen, setTrue, setFalse] = useBooleanState(defaultOpen);

    return (
        <>
            <button onClick={setTrue}>Open Modal</button>
            <Modal 
                isOpen={isOpen} 
                onClose={() => {
                    setFalse();
                    onClose();
                }}
            >
                <ModalTitle>Test Modal</ModalTitle>
                <ModalContent>Test Content</ModalContent>
            </Modal>
        </>
    )
}

beforeEach(() => {
    onClose.mockReset();
})


test("Renders nothing if isOpen is false", () => {
    render(<Modal isOpen={false} onClose={vi.fn()} />);

    expect(screen.queryByRole("dialog")).not.toBeInTheDocument()
});

test("Renders a dialog if isOpen is true using showModal", () => {
    vi.spyOn(HTMLDialogElement.prototype, "showModal");

    render(<Modal isOpen={true} onClose={vi.fn()} />);

    expect.soft(screen.getByRole("dialog")).toBeInTheDocument()
    expect.soft(HTMLDialogElement.prototype.showModal).toHaveBeenCalledOnce()
})

test("When isOpen is changed, the dialog is shown or hidden", () => {
    render(<TestComponent />);

    expect.soft(screen.queryByRole("dialog")).not.toBeInTheDocument()

    act(() => {
        screen.getByText("Open Modal").click()
    })

    expect.soft(screen.getByRole("dialog")).toBeInTheDocument()
    expect.soft(screen.getByText("Test Modal")).toBeInTheDocument()
    expect.soft(screen.getByText("Test Content")).toBeInTheDocument()
})

test("When the dialog's backdrop is clicked, onClose is called", () => {
    render(<TestComponent defaultOpen={true}/>);

    expect.soft(screen.getByRole("dialog")).toBeInTheDocument()

    act(() => {
        fireEvent.mouseDown(screen.getByRole("dialog"))
        fireEvent.mouseUp(screen.getByRole("dialog"))
    })

    expect.soft(screen.queryByRole("dialog")).not.toBeInTheDocument()
    expect.soft(onClose).toHaveBeenCalledOnce()
})

test("When the dialog's canceled event is fired, onClose is called", () => {
    render(<TestComponent defaultOpen={true}/>);

    expect.soft(screen.getByRole("dialog")).toBeInTheDocument()

    act(() => {
        screen.getByRole("dialog").dispatchEvent(new Event("cancel"))
    })

    expect.soft(screen.queryByRole("dialog")).not.toBeInTheDocument()
    expect.soft(onClose).toHaveBeenCalledOnce()
})

test("If the mouse was dragged to the backdrop, onClose is not called", () => {
    render(<TestComponent defaultOpen={true}/>);

    expect.soft(screen.getByRole("dialog")).toBeInTheDocument()

    act(() => {
        fireEvent.mouseDown(screen.getByText("Test Content"))
        fireEvent.mouseUp(screen.getByRole("dialog"))
    })

    expect.soft(screen.getByRole("dialog")).toBeInTheDocument()
    expect.soft(onClose).not.toHaveBeenCalled()
})

describe("Helpers", () => {
    test("ModalContent renders a div and matches snapshot", () => {
        const comp = render(<ModalContent>Test Content</ModalContent>);

        expect.soft(screen.getByText("Test Content").tagName).toBe("DIV")
        expect.soft(comp).toMatchSnapshot()
    })

    test("ModalTitle renders an h1 and matches snapshot", () => {
        const comp = render(<ModalTitle>Test Title</ModalTitle>);

        expect.soft(screen.getByText("Test Title").tagName).toBe("H1")
        expect.soft(comp).toMatchSnapshot()
    })
})

describe("Dialog style", () => {
    test("Matches idle snapshot", () => {
        const comp = render(<Dialog stage="idle"/>);

        expect(comp).toMatchSnapshot()
    })

    test("Matches enter snapshot", () => {
        const comp = render(<Dialog stage="enter"/>);

        expect(comp).toMatchSnapshot()
    })
})

describe("Responsive", () => {
    test("By default, the width if 500px with no max-height", () => {
        render(<Modal isOpen={true} onClose={vi.fn()} />)

        expect.soft(screen.getByRole("dialog")).toHaveStyle({ width: "500px", maxHeight: "" })
    })

    test("Small screens have a set max-width, max-height, and are aligned to the bottom", () => {
        render(<Modal isOpen={true} onClose={vi.fn()} />)

        const dialog = screen.getByRole("dialog");

        expect.soft(dialog).toHaveStyleRule("max-width", "100%", { media: mediaQueries.max("small") })
        expect.soft(dialog).toHaveStyleRule("max-height", "75%", { media: mediaQueries.max("small") })
        expect.soft(dialog).toHaveStyleRule("margin", "auto auto 0px auto", { media: mediaQueries.max("small") })    
        expect.soft(dialog).toHaveStyleRule("border-radius", `${vars.borderRadius} ${vars.borderRadius} 0px 0px`, { media: mediaQueries.max("small") })
    })
})