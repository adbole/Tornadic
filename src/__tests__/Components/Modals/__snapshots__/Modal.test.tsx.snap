// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dialog style > Matches enter snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: inherit;
  margin: auto;
  width: 500px;
  background-color: var(--background);
  border-radius: var(--border-radius);
  border: none;
  padding: 0;
  overflow: hidden;
  -webkit-transform: translateY(100vh);
  -moz-transform: translateY(100vh);
  -ms-transform: translateY(100vh);
  transform: translateY(100vh);
  -webkit-transition: -webkit-transform 1s ease;
  transition: transform 1s ease;
  -webkit-transform: translateY(0);
  -moz-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.emotion-0::backdrop {
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  background-color: rgba(0, 0, 0, 0.5);
  opacity: 0;
  -webkit-transition: opacity 1s ease;
  transition: opacity 1s ease;
  cursor: pointer;
}

@media screen and (max-width: 800px) {
  .emotion-0 {
    max-width: 100%;
    max-height: 75%;
    margin: auto auto 0px auto;
    border-radius: var(--border-radius) var(--border-radius) 0px 0px;
  }
}

.emotion-0::backdrop {
  opacity: 1;
}

<div>
  <dialog
    class="emotion-0 emotion-1"
  />
</div>
`;

exports[`Dialog style > Matches idle snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: inherit;
  margin: auto;
  width: 500px;
  background-color: var(--background);
  border-radius: var(--border-radius);
  border: none;
  padding: 0;
  overflow: hidden;
  -webkit-transform: translateY(100vh);
  -moz-transform: translateY(100vh);
  -ms-transform: translateY(100vh);
  transform: translateY(100vh);
  -webkit-transition: -webkit-transform 1s ease;
  transition: transform 1s ease;
}

.emotion-0::backdrop {
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  background-color: rgba(0, 0, 0, 0.5);
  opacity: 0;
  -webkit-transition: opacity 1s ease;
  transition: opacity 1s ease;
  cursor: pointer;
}

@media screen and (max-width: 800px) {
  .emotion-0 {
    max-width: 100%;
    max-height: 75%;
    margin: auto auto 0px auto;
    border-radius: var(--border-radius) var(--border-radius) 0px 0px;
  }
}

<div>
  <dialog
    class="emotion-0 emotion-1"
  />
</div>
`;

exports[`Helpers > ModalContent renders a div and matches snapshot 1`] = `
.emotion-0 {
  overflow-y: auto;
  padding: 20px;
}

.emotion-0 h2,
.emotion-0 h3,
.emotion-0 h4,
.emotion-0 h5,
.emotion-0 h6 {
  font-weight: 500;
}

.emotion-0 hr {
  height: 1px;
  background-color: #efdddd57;
  border: none;
  margin: 5px;
}

.emotion-0>p {
  margin-bottom: 10px;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    Test Content
  </div>
</div>
`;

exports[`Helpers > ModalTitle renders an h1 and matches snapshot 1`] = `
.emotion-0 {
  text-align: center;
  padding: 20px;
}

.emotion-0>* {
  background: transparent;
  border: none;
  color: inherit;
  font-size: inherit;
}

.emotion-0>*:focus {
  outline: none;
}

<div>
  <h1
    class="emotion-0 emotion-1"
  >
    Test Title
  </h1>
</div>
`;
